richtext/plugins/org.polarsys.kitalpha.richtext.widget.tools/src/org/polarsys/kitalpha/richtext/widget/tools/dialogs/MDEAddLinkDialog.java

```diff
@@ -11,237 +11,44 @@
-    GridData comboGridData = new GridData(GridData.FILL_HORIZONTAL);
-    comboGridData.horizontalSpan = 2;
-    linkTypeCombo.setLayoutData(comboGridData);
-
-    final String[] itemsLinkList = linkManager.getAllLinkLabels().toArray(new String[0]);
-
-    linkTypeCombo.setItems(itemsLinkList);
-    linkType = itemsLinkList[0];
-    linkTypeCombo.setText(linkType);
-    linkTypeCombo.addSelectionListener(new SelectionListener() {
-      @Override
-      @SuppressWarnings("synthetic-access")
-      public void widgetSelected(SelectionEvent event) {
-        linkType = itemsLinkList[linkTypeCombo.getSelectionIndex()];
-        browseButton.setEnabled(!Constants.URL_LABEL.equals(linkType));
-        urlLabel.setEnabled(Constants.URL_LABEL.equals(linkType) || Constants.FILE_LABEL.equals(linkType)
-            || Constants.FILE_LOCAL_LABEL.equals(linkType));
-        urlText.setEnabled(Constants.URL_LABEL.equals(linkType) || Constants.FILE_LABEL.equals(linkType)
-            || Constants.FILE_LOCAL_LABEL.equals(linkType));
-        if (Constants.URL_LABEL.equals(linkType)) {
-          urlDisplayNameText.setText(richText.getSelectedText());
-        } else {
-          urlDisplayNameText.setText(""); //$NON-NLS-1$
-        }
-        urlText.setText(""); //$NON-NLS-1$
-        updateHeaderMessages();
-      }
-
-      @Override
-      public void widgetDefaultSelected(SelectionEvent event) {
-        // do nothing
-      }
-    });
-
-    linkTypeCombo.moveAbove(urlLabel);
-  }
-
-  private void createDisplayTextURL(Composite composite) {
-
-    urlDisplayNameLabel = new Label(composite, SWT.NONE);
-    urlDisplayNameLabel.setText(Messages.RichTextWidget_Dialog_Add_Link_URL_Display);
-
-    urlDisplayNameText = new Text(composite, SWT.BORDER);
-    GridDataFactory.fillDefaults().grab(true, false).hint(300, SWT.DEFAULT).applyTo(urlDisplayNameText);
-
-    Label linkTypeLabel = new Label(composite, SWT.NONE);
-    linkTypeLabel.setText(Messages.RichTextWidget_Dialog_Add_Link);
-    linkTypeLabel.moveAbove(urlLabel);
-  }
-
-  private void createUrlLabel(Composite composite) {
-    urlLabel = new Label(composite, SWT.NONE);
-    urlLabel.setText("URL");
-    urlText = new Text(composite, SWT.BORDER);
-    GridDataFactory.fillDefaults().grab(true, false).hint(300, SWT.DEFAULT).applyTo(urlText);
-    urlText.addModifyListener(new ModifyListener() {
-      @Override
-      public void modifyText(ModifyEvent e) {
-        if (getOK() != null) {
-          getOK().setEnabled(urlText.getText().trim().length() > 0);
-        }
-      }
-    });
+  @Override
+  protected void okPressed() {
+    super.okPressed();
+    richText.forceFocus();
   }
 
   @Override
-  protected void createButtonsForButtonBar(Composite parent) {
-    createButton(parent, IDialogConstants.OK_ID, IDialogConstants.OK_LABEL, true);
-    createButton(parent, IDialogConstants.CANCEL_ID, IDialogConstants.CANCEL_LABEL, false);
+  protected void setEncodedURL(String url) {
```

Comment:
Hm as a second thought I'm not even sure if it's a good idea to inherit from the MDELinkDialog for this richtext use-case. Would it be cleaner to just call the MDELinkDialog here, get back the url, insert it to the richtext and focus on it. Other custo such as dialog name could be directly passed in the constructor of the dialog.